### PR TITLE
Migrate property getters to new signature for `Value::get`

### PR DESCRIPTION
**Pull request by fengalin**
... in consequence of https://github.com/gtk-rs/glib/pull/513

### CODE REVIEW

File: src/codegen/property_body.rs [hunk 1]
```diff
@@ -138,7 +138,10 @@ impl Builder {
         } else {
             ".unwrap()"
         };
-        body.push(Chunk::Custom(format!("value.get(){}", unwrap)));
+        body.push(Chunk::Custom(format!(
+            "value.get().expect(\"Return Value for property getter\"){}",
```
Comment (sdroege) [inline src/codegen/property_body.rs]: Maybe include the property name here and why this fails

Comment (fengalin) [reply src/codegen/property_body.rs]: Ok for name. However, it though the [error would be displayed in the panic message](https://doc.rust-lang.org/std/result/enum.Result.html#method.expect), so we only needed to write the context in the `expect` message.

Comment (sdroege) [reply src/codegen/property_body.rs]: true

Comment (fengalin) [reply src/codegen/property_body.rs]: ... but unfortunately, `expect` seems to be using the `Debug` version, not the `Display` one: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=7da532d4cd23ff02e61fb7085df8bab8

So we would get (example from `gio`):

```
thread 'main' panicked at 'Return Value for property `from-charset` getter: GetError { actual: guint, requested: gboolean }'
```

We loose the `TypeMismatch` part even though it is kind of obvious from the fields. So I guess the `enum` version could be applicable finally:

``` rust
pub enum GetError {
    TypeMismatch { actual: Type, requested: Type },
}
```

Comment (sdroege) [reply src/codegen/property_body.rs]: :+1:

Comment (sdroege) [reply src/codegen/property_body.rs]: OTOH, how does it work for `TryFromIntError` then? That should have the same problem you have here.

Comment (fengalin) [reply src/codegen/property_body.rs]: Not much details: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=0c0d17053f263ea0bb6b2c6cb2340847

Comment (sdroege) [reply src/codegen/property_body.rs]: Let's keep consistent with that then, I guess

Comment (sdroege) [reply src/codegen/property_body.rs]: I.e. it's not our problem but something that should be solved in `expect()` and `unwrap()`

Comment (fengalin) [reply src/codegen/property_body.rs]: All right!

**Comment by sdroege** [comment]
Is this all ready from your side then? :) I'd give it another review later

**Comment by fengalin** [comment]
> Is this all ready from your side then?

Not yet, I'm checking the other crates now :)

**Comment by fengalin** [comment]
It should be good to go now. So far, I checked:

- `gtk-rs/gtk`: generation & unit tests with features `subclassing,futures` (which also pulls `atk`, `gdk`, `gdk-pixbuf` & `gio`)
- `gstreamer-rs/gstreamer`: generation & unit tests with features `subclassing,futures,serde`.

**Comment by fengalin** [comment]
Guys, when you get time, could you review & merge this one please? This will allow me to regen the other crates with the actual `gir` version.

File: src/codegen/property_body.rs [hunk 2]
```diff
@@ -138,7 +138,10 @@ impl Builder {
         } else {
             ".unwrap()"
         };
-        body.push(Chunk::Custom(format!("value.get(){}", unwrap)));
+        body.push(Chunk::Custom(format!(
+            "value.get().expect(\"Return Value for property `{}` getter\"){}",
```
Comment (EPashkin) [inline src/codegen/property_body.rs]: This error not tell about actual reason.
IMHO just `unwrap` give better error.

Comment (fengalin) [reply src/codegen/property_body.rs]: Yes its does... somewhat. See this discussion: https://github.com/gtk-rs/gir/pull/825#pullrequestreview-272474063 (you will need to click "Show resolved").

Comment (EPashkin) [reply src/codegen/property_body.rs]: Sorry, I missed that discussion.

**Comment by EPashkin** [comment]
@fengalin Thanks, I will merge then

**Comment by fengalin** [comment]
Thanks @EPashkin

**Pull request merged**